examples/inspection/plot_partial_dependence.py

```diff
@@ -213,3 +225,61 @@
 plt.subplots_adjust(top=0.9)
 
 plt.show()
+
+##############################################################################
+# Individual Conditional Expectation (ICE) Plots
+# ----------------------------------------------
+#
+# One of the main limitations of ICE plots is that the plot gets overcrowded
+# if many ICE curves are drawn. Due to this, we will use a random sample of
+# 50 instances for the ICE plot.
+#
+# ICE plots are achieved by setting ``individual=True``.
+# Let's now compute the ICE plots for this Gradient Boosting model:
+
+X_train_sample = X_train.sample(50, random_state=0)
+
+features = ['MedInc', 'AveOccup', 'HouseAge', 'AveRooms']
+
+print('Computing ICE plots...')
+plot_partial_dependence(est, X_train_sample, features, n_jobs=3,
+                        grid_resolution=20, individual=True)
+fig = plt.gcf()
+fig.suptitle('ICE of house value on non-location features\n'
+             'for the California housing dataset, with Gradient Boosting')
+fig.subplots_adjust(hspace=0.3)
+
+plt.show()
+
+##############################################################################
+# In ICE plots it might not be easy to see the average effect of the 'target'
+# variable. Hence, it is recommended to use ICE plots together with partial
+# dependency plots. They can be plotted in the same plot with ``individual``
+# parameter set to ``'both'``.
+
+print('Computing ICE and PD plots...')
+plot_partial_dependence(est, X_train_sample, features, n_jobs=3,
+                        grid_resolution=20, individual='both')
+fig = plt.gcf()
+fig.suptitle('ICE and PD of house value on non-location features\n'
+             'for the California housing dataset, with Gradient Boosting')
+fig.subplots_adjust(hspace=0.3)
+
+plt.show()
+
+##############################################################################
+# Analysis of the ICE plots
+# -------------------------
+#
+# From the partial dependence plots, we can see that the median house price
+# increases linearly with the median income (top left) and that the median
+# house price drops when the average occupants per household increases
+# (top middle). However, from  ICE plots we can see that there are some
+# exceptions, where the house price remain constant with median income and
+# average occupants.
```

Comment:
While I agree with the interpretation, I am thinking if it would be worth to mention that we are creating potential meaningless due to the correlation with another features. WDYT?

Reply:
Yes, that makes sense.

Reply:
See https://github.com/scikit-learn/scikit-learn/pull/16619/commits/5cc47a41bf4fa0bf23c798703e089675c66a2784